test/conformance/envpropagation_test.go

```diff
@@ -53,27 +55,47 @@ func TestSecretsFromEnv(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-
 	t.Logf("Successfully created test secret: %v", secret)
 
-	err = fetchEnvironmentAndVerify(t, clients, corev1.EnvVar{
-		Name: testKey,
-		ValueFrom: &corev1.EnvVarSource{
-			SecretKeyRef: &corev1.SecretKeySelector{
+	defer func() {
+		if err := cleanupSecret(secretName, clients); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	t.Run("env", func(t *testing.T) {
+		err = fetchEnvironmentAndVerify(t, clients, WithEnv(corev1.EnvVar{
```

Comment:
t.Parallel() here and on line 84 will let this run parallel with each other

Reply:
I originally had that, but it detaches and the outer test returns cleaning up the secret / configmap. :(

Reply:
We could add synchronization, but these are super fast as-is, so I biased for clarity.